src/components/Search/SearchPageHeader.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useCallback} from 'react';
+import ButtonWithDropdownMenu from '@components/ButtonWithDropdownMenu';
+import type {DropdownOption} from '@components/ButtonWithDropdownMenu/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import * as Expensicons from '@components/Icon/Expensicons';
+import * as Illustrations from '@components/Icon/Illustrations';
+import useLocalize from '@hooks/useLocalize';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as SearchActions from '@libs/actions/Search';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import type {SearchQuery, SelectedTransactions} from '@src/types/onyx/SearchResults';
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import type IconAsset from '@src/types/utils/IconAsset';
+
+type SearchHeaderProps = {
+    query: SearchQuery;
+    selectedItems?: SelectedTransactions;
+    clearSelectedItems?: () => void;
+    hash: number;
+};
+
+type SearchHeaderOptionValue = DeepValueOf<typeof CONST.SEARCH_BULK_ACTION_TYPES> | undefined;
+
+function SearchPageHeader({query, selectedItems = {}, hash, clearSelectedItems}: SearchHeaderProps) {
+    const {translate} = useLocalize();
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {isSmallScreenWidth} = useResponsiveLayout();
+    const headerContent: {[key in SearchQuery]: {icon: IconAsset; title: string}} = {
+        all: {icon: Illustrations.MoneyReceipts, title: translate('common.expenses')},
+        shared: {icon: Illustrations.SendMoney, title: translate('common.shared')},
+        drafts: {icon: Illustrations.Pencil, title: translate('common.drafts')},
+        finished: {icon: Illustrations.CheckmarkCircle, title: translate('common.finished')},
+    };
+
+    const getHeaderButtons = useCallback(() => {
```

Comment:
```suggestion
    const getSelection(Options|Buttons) = useCallback(() => {
```

Reply:
IThis is the convention used in other list components: `WorkspaceCategoriesPage`, `WorkspaceTagsPage` etc.